tests/meltano/cli/test_select.py

```diff
@@ -33,3 +39,40 @@ def test_update_select_pattern(self, cli_runner, tap):
         assert_cli_runner(result)
         json_config = json.loads(result.stdout)
         assert "mock.*" not in json_config["_select"]
+
+    @pytest.mark.usefixtures("project")
+    def test_select_list(self, cli_runner, tap, monkeypatch: pytest.MonkeyPatch):
+        async def mock_list_all(*args, **kwargs):  # noqa: ARG001
+            result = ListSelectedExecutor()
+            result.streams = {
+                SelectedNode(key="users", selection=SelectionType.SELECTED)
+            }
+            result.properties = {
+                "users": {
+                    SelectedNode(key="id", selection=SelectionType.SELECTED),
+                    SelectedNode(key="name", selection=SelectionType.EXCLUDED),
+                }
+            }
+            return result
+
+        monkeypatch.setattr(
+            SelectService,
+            "list_all",
+            mock_list_all,
+        )
+
+        # list selection
+        result = cli_runner.invoke(
+            cli,
+            [
+                "--no-environment",
+                "select",
+                tap.name,
+                "--list",
+                "--all",
+            ],
+        )
+        assert_cli_runner(result)
+
+        assert "[SelectionType.SELECTED] users.id" in result.stdout
+        assert "[SelectionType.EXCLUDED] users.name" in result.stdout
```

Comment:
Turns out this uncovered something I hadn't noticed before 🙃 

https://github.com/meltano/meltano/issues/8526

Reply:
Fixed by https://github.com/meltano/meltano/issues/8527